component/otelcol/config_filter_test.go

```diff
@@ -0,0 +1,172 @@
+package otelcol_test
+
+import (
+	"testing"
+
+	"github.com/grafana/agent/component/otelcol"
+	"github.com/stretchr/testify/require"
+)
+
+func TestConvertMatchProperties(t *testing.T) {
+	version := "3.8.0"
```

Comment:
Idk if there is a better way to do this inline at `Version: &version,`?

Reply:
You can use [pointer.String](https://pkg.go.dev/k8s.io/utils/pointer#String) from k8s.io/utils/pointer.